tests/publiccloud/check_registercloudguest.pm

```diff
@@ -106,14 +105,16 @@ sub run {
     if (is_byos()) {
         $instance->ssh_assert_script_run(cmd => 'sudo SUSEConnect --version');
         $instance->ssh_assert_script_run(cmd => "sudo SUSEConnect $regcode_param");
-        $instance->ssh_assert_script_run(cmd => "sudo ${path}registercloudguest --clean");
+        cleanup_instance($instance);
     }
 
-    $instance->ssh_script_retry(cmd => "sudo ${path}registercloudguest $regcode_param", timeout => 300, retry => 3, delay => 120);
-    check_instance_registered($instance);
+    new_registration($instance);
```

Comment:
This new routine call doesn't check `is_byos` neither before the call nor in its body: is this expected?

Reply:
yes , it is expected . because for PAYG we still want : 
1. execute `registercloudguest` without regcode to make check that `registercloudguest` execution would not do any harm to PAYG instance 
2.  obviously we want to run `check_instance_registered`  because both checks  inside this function expected to work on PAYG image the same way as they work on registered BYOS image

Reply:
ok

Reply:
why ? this function is called ONLY for BYOS

Reply:
Well: if here  `is_byos is false`, then `new_registration` will run `registercloudguest` without parameters: if this is expected, then all ok.